pkg/extensionapis/walrus/connector.go

```diff
@@ -19,22 +20,76 @@ import (
 type ConnectorHandler struct {
 	extensionapi.ObjectInfo
 	extensionapi.CurdOperations
+
+	Client ctrlcli.Client
 }
 
 func (h *ConnectorHandler) SetupHandler(
 	ctx context.Context,
 	opts extensionapi.SetupOptions,
 ) (gvr schema.GroupVersionResource, srs map[string]rest.Storage, err error) {
+	// Configure field indexer.
+	fi := opts.Manager.GetFieldIndexer()
+	err = fi.IndexField(ctx, &walruscore.Connector{}, "metadata.name",
+		func(obj ctrlcli.Object) []string {
+			if obj == nil {
+				return nil
+			}
+			return []string{obj.GetName()}
+		})
+	if err != nil {
+		return schema.GroupVersionResource{}, nil, err
+	}
+
 	// Declare GVR.
 	gvr = walrus.SchemeGroupVersionResource("connectors")
 
+	// Create table convertor to pretty the kubectl's output.
+	var tc rest.TableConvertor
+	{
+		tc, err = extensionapi.NewJSONPathTableConvertor(
+			extensionapi.JSONPathTableColumnDefinition{
+				TableColumnDefinition: meta.TableColumnDefinition{
+					Name: "Category",
+					Type: "string",
+				},
+				JSONPath: ".spec.category",
+			},
+			extensionapi.JSONPathTableColumnDefinition{
+				TableColumnDefinition: meta.TableColumnDefinition{
+					Name: "Type",
+					Type: "string",
+				},
+				JSONPath: ".spec.type",
+			},
+			extensionapi.JSONPathTableColumnDefinition{
+				TableColumnDefinition: meta.TableColumnDefinition{
+					Name: "Applicable Environment Type",
+					Type: "string",
+				},
+				JSONPath: ".spec.applicable_environment_type",
```

Comment:
Should update when applicable_environment_type update to applicableEnvironmentType